### PR TITLE
Update throttling for EPC

### PR DESCRIPTION
**Pull request by wpscholar**

### CODE REVIEW

File: endurance-page-cache.php [hunk 1]
```diff
@@ -585,19 +595,50 @@ public function purge_cdn_single( $pattern = '' ) {
...
-				return true;
+		public function should_throttle( $uri ) {
+
+			$should_throttle = false;
+
+			$this->should_update_purges = true;
+
+			$hash = md5( $uri );
+
+			if ( array_key_exists( $hash, $this->purged ) ) {
```
Comment (0aveRyan) [inline endurance-page-cache.php]: It's kinda a micro-optimization, but `isset()` is [typically faster](https://gist.github.com/alcaeus/536156663fac96744eba77b3e133e50a) than `array_key_exists()` and spreading that across the farm and how frequently this function is called will add up.

File: endurance-page-cache.php [hunk 2]
```diff
@@ -585,19 +595,50 @@ public function purge_cdn_single( $pattern = '' ) {
...
+		 */
+		public function shutdown() {
+			if ( $this->should_update_purges ) {
+				$purged = array_filter( $this->purged, array( $this, 'filter_expired_timestamps' ) );
+				set_transient( 'epc_purged', $purged, 60 );
+			}
+		}
+
+		/**
+		 * Filters an array of timestamp values and removes items that have expired.
```
Comment (0aveRyan) [inline endurance-page-cache.php]: Is this code comment meant for another function? It's how this helper is used, but not what this method is actually doing -- testing timestamp against present.

File: endurance-page-cache.php [hunk 3]
```diff
@@ -585,19 +595,50 @@ public function purge_cdn_single( $pattern = '' ) {
...
+					unset( $this->purged[ $hash ] );
+				} else {
+					$should_throttle = true;
+				}
 			}
-			set_transient( 'epc_purged_' . md5( $value ), time(), 60 );
-			$this->purged[] = md5( $value );
 
-			return false;
+			$this->purged[ $hash ] = time() + 60;
```
Comment (earnjam) [inline endurance-page-cache.php]: This is always going to push the hash time out by 60 seconds, whether it's been throttled or not. So as long as these purge requests keep firing within the minute it keeps pushing them out. 

Should we have a hard 60 second cap, and allow purges after that, or keep pushing it out like this every time if a purge fires within that limit?

File: endurance-page-cache.php (hunk 2 above)
Comment (earnjam) [reply endurance-page-cache.php]: Agreed. Should probably move this comment up above to line 628 and then clarify what this function specifically does.

**Comment by wpscholar** [comment]
@0aveRyan @earnjam - PR has been updated. Please re-review.

**Pull request closed without merging**